lms/static/js/spec/discovery/discovery_factory_spec.js

```diff
@@ -194,5 +195,17 @@ define([
             $('.search-facets li [data-value="edX1"]').trigger('click');
             expect($('.active-filter [data-value="edX1"]').length).toBe(0);
         });
+
+        it('filters by default language', function() {
+            DiscoveryFactory(MEANINGS, '', 'en', 'Asia/Kolkata', true);
+            var requests = AjaxHelpers.requests(this);
+            $('.discovery-submit').trigger('click');
+            AjaxHelpers.respondWithJson(requests, JSON_RESPONSE);
+            expect($('.courses-listing article').length).toEqual(1);
+            expect($('.courses-listing .course-title')).toContainHtml('edX Demonstration Course');
+            expect($('.active-filter').length).toBe(1);
+            expect($('.active-filter .query')).toContainHtml("English");
+        });
```

Comment:
I noticed that when my preferred language doesn't match any of my course's languages that I see header text like this:

> We couldn't find any results for ""

which I think students may find confusing, because the text is confusing, and because the student didn't actually perform an explicit search.

An easy way around this would be to avoid showing that header if there's no search text.

Reply:
> Since there's only ever one course in the returned list, this test doesn't actually verify that the filter was applied.
Could you add another course that uses a different language so we can see that this filter actually applies?

@pomegranited Actually the test should not be about checking API response as it is mocked above by `AjaxHelpers.respondWithJson(requests, JSON_RESPONSE);`. So I modified to check the request body as it should include language filter automatically when default filter is enabled.

> An easy way around this would be to avoid showing that header if there's no search text.

Makes sense. Done :+1: